[Discover] Rename Saved Search to Discover Session (#202217)
elastic/kibana: docs/concepts/save-query.asciidoc

Saved queries are different than <<save-open-search,saved searches>>,
which include the *Discover* configuration&mdash;selected columns in the document table, sort order, and
{data-source}&mdash;in addition to the query.

> Saved searches, known as Discover Sessions, are primarily used for adding search results to a dashboard.

Author: I think it's okay to keep the paragraph. I updated it with "Discover session" mentions.

> Saved searches are primarily used for adding search results to a dashboard. In UI saved searches are now called Discover Sessions.

Reviewer: ```suggestion
Saved searches, known as Discover Sessions, are primarily used for adding search results to a dashboard.
```
